analyses/survival-analysis/survival-analysis_histology.Rmd

```diff
@@ -0,0 +1,202 @@
+for(ind_var in c("tp53_score", "telomerase_score")){
+  model <- paste0("survival::Surv(OS_days, OS_status_fixed) ~ ", ind_var)
+  
+  fit <- survival::coxph(
+        formula(model),
+        data = meta_indep
+      )
+  # generate output
+  table <- broom::tidy(fit) 
+  
+  # Save the table data in a TSV
+  readr::write_tsv(table, file.path(results_dir, paste0("cox_reg_results_per_", ind_var, ".tsv")))
+  
+  print(table)
+}
+
+```
+
+## Log Rank analysis 
+
+### Categorical variable cancer group and HGG or not 
+```{r}
+# separate into HGG or non-HGG
+meta_indep <- meta_indep %>% 
+  dplyr::mutate(hgg_group = case_when(
+    short_histology == "HGAT" ~ "HGAT",
+    short_histology != "HGAT" ~ "non-HGAT"
+  ))
+
+# filter to contain only samples in short histology that that have >=10 samples
+meta_indep_sh_filtered <- meta_indep %>% 
+  dplyr::group_by(short_histology) %>% 
+  dplyr::mutate(n_cg = n()) %>% 
+  dplyr::filter(n_cg >=10) 
+
+```
+
+### Generate output for categorical files 
+```{r}
+ind_var_list <- c("hgg_group", "short_histology")
+meta_data_list <- list(meta_indep, meta_indep_sh_filtered)
+
+for(i in 1:length(ind_var_list)){
+  ind_var <- ind_var_list[i]
+  meta_data <- meta_data_list[[i]]
+  
+  # define model
+  model <- paste0("survival::Surv(OS_days, OS_status_fixed) ~ ", ind_var)
+  
+  # run survival analysis
+  fit <- survival::survdiff(formula(model),
+                            data = meta_data)
+  # Obtain p value for Chi-Squared stat
+  fit$p.value <- pchisq(fit$chisq, df = 1, lower = FALSE)
+  
+  print(fit)
+  # save the output
+  saveRDS(fit, file.path(results_dir, paste0("log_rank_survival_per_", ind_var, ".RDS")))
+  
+  # generate plots fit
+  fit_plot <- survfit(formula(model), data = meta_data)
+  plot_coxph <- survminer::ggsurvplot(fit_plot,
+                                      data=meta_data,
+                                      xlim = c(0, 5000),
+                                      break.time.by = 500,
+                                      pval = TRUE, 
+                                      conf.int = TRUE,
+                                      risk.table = TRUE, # Add risk table
+                                      risk.table.col = "strata", # Change risk table color by groups
+                                      linetype = "strata", # Change line type by groups
+                                      surv.median.line = "hv", # Specify median survival
+                                      ggtheme = theme_bw()) # Change ggplot2 theme
+  
+  # Make this plot a combined plot
+  surv_plot_coxph <- cowplot::plot_grid(plot_coxph[[1]], 
+                                        plot_coxph[[2]], 
+                                        nrow = 2, 
+                                        rel_heights = c(2.5, 2.5))
+  
+  print(surv_plot_coxph)
+  
+  # Save the plot
+  cowplot::save_plot(filename = file.path(plots_dir, paste0("coxph_survival_per_", ind_var, ".png")), plot = surv_plot_coxph)
```

Comment:
Although I don't think we should worry much about polishing these plots, I think we should aim to get these to a point where they can be interpreted by (pull request) reviewers. They are a little smushed at the moment – I think changing the width and height could go a long way, but I might also try not making this a combined figure.

Reply:
Yes - I have now split them into two plots, move the legends to the right for the survival plots and remove color legend for the risk table. Still not perfect but now we can see all the legends.